feat: Design a database and implement a data structure

abrizu opened the pull request:
The pull request is to add a set of features including designing a database and implementing a data structure. These features allow students to store data, such as grades, within a student database and calculate each assignment / test, as well as the weight it holds on their final grade.

The changes are:
1. Adding a new database component to store student data
2. Connecting to the frontend API
3. Adding a new calculator function to retrieve stored data 
4. Calculate student grades given assignment / test data

abrizu committed:
feat: Design a database regarding student grades

- Added a new component capable of independently storing student grades, making student grades accessible
- Linked the component function to the main application
- Connected the backend with the frontend API

Issue #: 01

abrizu committed:
feat:  Implement a data structure to calculate grades

- Added a function to calculate each assignment / test category
- Applied the function to student's grades in the backend
- Connected the system to the main HTML page "emich.edu/student-id/grades"
 
Issue #: 02

aabrizuelaa reviewed (approved):
Add a driver to manage the functions, other than that, the pull request is ready to be merged.

abrizu committed:
Commit after review

- This commit confirms the approval to merge the branches.
- Added Driver() component to run each function individually.

helloammaro commented:
Will this include a site where the student can check / calculate grades? Instead of being implemented directly in the database.
